examples/SmoothedParticleHydrodynamics/src/model/functions.c

```diff
@@ -0,0 +1,233 @@
+
+    return 0;
+}
+
+/**
+ * computeDensityPressure FLAMEGPU Agent Function
+ * Automatically generated using functions.xslt
+ * @param agent Pointer to an agent structure of type xmachine_memory_FluidParticle. This represents a single agent instance and can be modified directly.
+ * @param location_messages  location_messages Pointer to input message list of type xmachine_message__list. Must be passed as an argument to the get_first_location_message and get_next_location_message functions.* @param density_pressure_messages Pointer to output message list of type xmachine_message_density_pressure_list. Must be passed as an argument to the add_density_pressure_message function.
+ */
+__FLAME_GPU_FUNC__ int computeDensityPressure(xmachine_memory_FluidParticle* agent, xmachine_message_location_list* location_messages, xmachine_message_location_PBM* partition_matrix, xmachine_message_density_pressure_list* density_pressure_messages){
+
+	// Compute unmodified particle density
+	float density = 0.0f;
+    
+    xmachine_message_location* current_message = get_first_location_message(location_messages, partition_matrix, agent->x, agent->y, agent->z);
+	while (current_message)
+	{
+		// For each agent add weighted density contribution
+		glm::vec3 diff = glm::vec3(current_message->x - agent->x, current_message->y - agent->y, current_message->z - agent->z);
+		float distance = length(diff);
+		float w = computeW(distance, SMOOTHING_LENGTH);
+		density += PARTICLE_MASS * w;
+		
+		current_message = get_next_location_message(current_message, location_messages, partition_matrix);
+	}
+
+	// Set density and pressure for the particle - negative pressures are not used
+	agent->density = max(density, MINIMUM_PARTICLE_DENSITY);
+	agent->pressure = max(PRESSURE_COEFFICIENT * (agent->density - FLUID_REST_DENSITY), 0.0f);
+
+	// Output location/density/pressure
+    add_density_pressure_message(density_pressure_messages, agent->id, agent->density, agent->pressure, agent->x, agent->y, agent->z, agent->dx, agent->dy, agent->dz, agent->isStatic);
+
+    return 0;
+}
+
+/**
+ * computeForce FLAMEGPU Agent Function
+ * Automatically generated using functions.xslt
+ * @param agent Pointer to an agent structure of type xmachine_memory_FluidParticle. This represents a single agent instance and can be modified directly.
+ * @param density_pressure_messages  density_pressure_messages Pointer to input message list of type xmachine_message__list. Must be passed as an argument to the get_first_density_pressure_message and get_next_density_pressure_message functions.* @param force_messages Pointer to output message list of type xmachine_message_force_list. Must be passed as an argument to the add_force_message function.
+ */
+__FLAME_GPU_FUNC__ int computeForce(xmachine_memory_FluidParticle* agent, xmachine_message_density_pressure_list* density_pressure_messages, xmachine_message_density_pressure_PBM* partition_matrix){
+    
+	glm::vec3 force = glm::vec3(0.0f);
```

Comment:
looks like a mix of tabs and spaces for indentation? 

Could you convert  to just one for consistency?